tests/phpunit/integration/Core/Authentication/User_Input_StateTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * User_Input_StateTest
+ *
+ * @package   Google\Site_Kit\Tests\Core\Authentication
+ * @copyright 2019 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Tests\Core\Authentication;
+
+use Google\Site_Kit\Tests\TestCase;
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Authentication\User_Input_State;
+use Google\Site_Kit\Core\Storage\User_Options;
+
+/**
+ * @group Authentication
+ */
+class User_Input_StateTest extends TestCase {
+
+	/**
+	 * User Input State instance
+	 *
+	 * @var User_Input_State
+	 */
+	protected $user_input_state;
+
+	/**
+	 * Set Up Test.
+	 */
+	public function setUp() {
+		$user_id                = $this->factory()->user->create();
+		$context                = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$this->user_options     = new User_Options( $context, $user_id );
+		$this->user_input_state = new User_Input_State( $this->user_options );
+	}
+
+	public function tearDown() {
+		unset( $this->user_input_state );
+	}
```

Comment:
I don't think this is needed, no need to unset the property. We only need to clean up things that have side effects.